### PR TITLE
Display short url on sign up form after successful signup

### LINKED ISSUE #142: Display SIgnup short URL on Sign Up Form
**Describe the solution you'd like**
Upon successful sign up, provide the signup-service's generated short link to the user. This is the same link sent to user via SMS and renders the same content as the email.

**Additional context**
We'll need to have the [signup service respond with the short link](https://github.com/OperationSpark/service-signups/issues/91) upon successful sign up. This will allow us to provide the link to the user in the UI.

### PR DESCRIPTION
**Pull request by ptbarnum4**
- [x] Fix unexpected input clear on enter
- [x] Dynamically set referredBy field(s) if `referred_by` query params exists. For example:
  - SNAP: `/programs/workforce/infoSession?referred_by=snap`
  - Google: `/programs/workforce/infoSession?referred_by=google`
  - Word of Mouth: `/programs/workforce/infoSession?referred_by=verbal@hallebot`
  - Custom: `/programs/workforce/infoSession?referred_by=AnyValue`
- [x] Display success message, info session date time, and renderer URL upon successful signup. Resolves #142 

https://github.com/OperationSpark/operationspark-org-website-2022/assets/37914211/e77f66fc-ab05-4f03-a41a-d698401c8ce6

### CODE REVIEW

File: src/Forms/Form.Workforce.tsx [hunk 1]
```diff
@@ -64,21 +68,22 @@ const WorkforceForm = ({ sessionDates, referredBy }: WorkforceFormProps) => {
...
-          title: 'Error',
-          msg: 'There was an error signing you up\nPlease reach out to us at "admissions@operationspark.org" or give us a call at 504-233-3838',
-        });
-      })
-      .finally(() => setIsSubmitting(false));
+    try {
+      const { data } = await axios.post('/api/infoSession/user', body);
+
+      form.notifySuccess({
+        msg: 'Info session registration for submitted. You should receive an email and text message shortly.',
```
Comment (harveysanders) [inline src/Forms/Form.Workforce.tsx:75]: Do you think we should dynamically add the "text message" part based on if they opted in for messaging?

File: src/Forms/Form.Workforce.tsx [hunk 2]
```diff
@@ -350,6 +364,31 @@ const WorkforceForm = ({ sessionDates, referredBy }: WorkforceFormProps) => {
...
+          </div>
+        ) : null}
+        {renderUrl && !isSubmitting ? (
+          <div className='form-overlay'>
+            <div className='form-complete-response'>
+              <h2>Success!</h2>
+              <p>
+                You have successfully registered for an info session on{' '}
+                <b className='primary-secondary'>{currentValues.sessionDate.name}</b>. You will
+                receive an email {currentValues.smsOptIn === 'true' ? 'and text message' : ''}{' '}
```
Comment (harveysanders) [inline src/Forms/Form.Workforce.tsx:380]: Yeah, like this one (see above comment)

**Review by harveysanders — APPROVED** [review]
Looks good. I just left a suggestion about the messaging

File: src/Forms/Form.Workforce.tsx (hunk 1 above)
Comment (ptbarnum4) [reply src/Forms/Form.Workforce.tsx:75]: good call

Comment (ptbarnum4) [reply src/Forms/Form.Workforce.tsx]: ### SMS `true`
Screenshot 2024-03-28 at 7 26 21 PM

### SMS `false`
Screenshot 2024-03-28 at 7 27 16 PM

**Comment by ptbarnum4** [comment]
oops

Screenshot 2024-03-28 at 7 59 44 PM
Screenshot 2024-03-28 at 7 59 48 PM

File: src/Forms/Form.Workforce.tsx (hunk 1 above)
Comment (ptbarnum4) [reply src/Forms/Form.Workforce.tsx]: - [x] Handle no session selected (future session) https://github.com/OperationSpark/operationspark-org-website-2022/pull/148/commits/9b01010d21347cfd5d3bddd7c75f77604b8c433e

File: src/Forms/Form.Workforce.tsx [hunk 3]
```diff
@@ -34,13 +34,15 @@ const WorkforceForm = ({ sessionDates, referredBy }: WorkforceFormProps) => {
   const currentValues = form.values();
 
   const handleSubmit = async () => {
+    if (isSubmitting) return;
+    setIsSubmitting(true);
```
Comment (harveysanders) [inline src/Forms/Form.Workforce.tsx:38]: Why do you need to call `setIsSubmitting(true)` if `isSubmitting` is already `true`?
```suggestion
```

**Review by harveysanders — APPROVED** [review]

File: src/Forms/Form.Workforce.tsx (hunk 3 above)
Comment (ptbarnum4) [reply src/Forms/Form.Workforce.tsx:38]: I'm not sure I understand the question.
- If `isSubmitting` is `true` -- `return`
- If `isSubmitting is `false` -- continue and set to true

Comment (harveysanders) [reply src/Forms/Form.Workforce.tsx:38]: ah yeah I misread the bracketless `if` statement

**Pull request merged**